plugin/external/pluginEntry.go

```diff
@@ -361,6 +433,15 @@ func (e *pluginEntry) List(ctx context.Context) ([]plugin.Entry, error) {
 
 	entries := make([]plugin.Entry, len(decodedEntries))
 	for i, decodedExternalPluginEntry := range decodedEntries {
+		if decodedExternalPluginEntry.TypeID == volumeFSTypeID {
+			var opts volumeFSOpts
```

Comment:
You could generalize this to "Check if user's asking for a core plugin entry" via a map of `<type_id> => <constructor>` where `<constructor>` could take the external plugin entry as its argument. Then this would read as:
```
if createEntry, ok := corePluginEntryTypeID(e.TypeID) {
  entry, err := createEntry(e)
  ...
}
// External plugin entry so rest of code is here
```

I think this generalization seems easy enough that we might as well do it here. Also helps clarify the code.

It might even be worth modifying the map to something like `<type_id>` => `<core_plugin_entry_info>` where `<core_plugin_entry_info>` includes the constructor and the corresponding schema. Then you can do a similar thing in EntrySchema.

**Note:** We should come up with a better name than core plugin entry so that it still persists even when some core plugins become separate Wash implementations . Maybe "wash-provided entry"? Don't really have good ideas here.

Reply:
Done.